### PR TITLE
Validate email on login and signup

### PR DESCRIPTION
**Pull request by timagixe**
**IMPORTANT:** cannot be merged to dev if [PR-75](https://github.com/BinaryStudioAcademy/bsa-2020-TaskyHusky/pull/75) is not merged yet.

_This PR includes lates dev + changes from PR-75 + code fixes of PR-75._

### CODE REVIEW

**Comment by timagixe** [comment]
Flow - [click me](https://trello.com/c/j8vrZnXz/126-fix-uppercase-email-on-registration#comment-5f34451083901c21479338b8)

File: client/src/helpers/fixEmail.helper.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const fixEmail: (txt: string) => string = (text) => text.trim().toLowerCase();
```
Comment (taras-dubyk) [inline client/src/helpers/fixEmail.helper.ts:1]: Can this be renamed to something like "normalizeEmail" or "normalizeString" ?

File: server/src/helpers/validateUserProfile.helper.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { validate } from 'class-validator';
```
Comment (taras-dubyk) [inline server/src/helpers/validateUserProfile.helper.ts:1]: Please create "validators" folder and move the file there

File: client/src/helpers/fixEmail.helper.ts (hunk 1 above)
Comment (timagixe) [reply client/src/helpers/fixEmail.helper.ts]: Done - 8a0350d594a9a495cbcd06d0270c664b7e85be9b

File: server/src/helpers/validateUserProfile.helper.ts (hunk 2 above)
Comment (timagixe) [reply server/src/helpers/validateUserProfile.helper.ts]: Done - acc5d71909f3cd9dbc7fbbea434599cd6313a000

**Pull request merged**